pkg/util/util.go

```diff
@@ -270,6 +271,26 @@ func DetectFileFormat(filePath string) (string, error) {
 	return "", fmt.Errorf("cannot find the file format in the output %s", output)
 }
 
+func GetImageVirtualSize(filePath string) (int64, error) {
+	type qemuImgInfo struct {
+		VirtualSize int64 `json:"virtual-size"`
+	}
```

Comment:
I just realized that we can make struct `QemuImgInfo` public, using one single function to fetch this virtual size as well image format info. Then we don't need the above function `DetectFileFormat` either.

Reply:
Good idea, I'll do that.